source/modulo_core/include/modulo_core/translators/message_readers.hpp

```diff
@@ -256,7 +258,7 @@ inline void safe_spatial_state_conversion(
  * referenced StateT instance can be modified as needed from the NewStateT value
  */
 template<typename StateT, typename NewStateT>
-inline void safe_joint_state_conversion(
+inline void safe_state_with_names_conversion(
```

Comment:
Changing the name of this function is technically API breaking change, considering that any user _might_ have included this header and call this function in their source code. It wouldn't be ABI breaking because any compiled targets would have inlined this function and would not need to look it up from the shared library. The risk is very small since it's only used locally here, but still.

I would prefer you mark a FIXME or TODO comment to patch before a 5.0 release, either keeping the old name and calling it for the io state case (easiest) or adding a new copied and renamed function / wrapper function just for io state as a temporary workaround

Reply:
Good catch